ide/rust/SyncRunner.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.vaticle.dependencies.ide.rust
+
+import com.vaticle.bazel.distribution.common.Logging.LogLevel.DEBUG
+import com.vaticle.bazel.distribution.common.Logging.LogLevel.ERROR
+import com.vaticle.bazel.distribution.common.Logging.Logger
+import com.vaticle.bazel.distribution.common.shell.Shell
+import com.vaticle.bazel.distribution.common.util.FileUtil.listFilesRecursively
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.ASPECTS
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.BAZEL
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.BAZEL_BIN
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.BUILD
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.INFO
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.OUTPUT_BASE
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.OUTPUT_GROUPS_RUST_IDE_SYNC
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.QUERY
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.RUST_IDE_SYNC_ASPECT
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.RUST_TARGETS_DEPS_QUERY
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.RUST_TARGETS_QUERY
+import com.vaticle.dependencies.ide.rust.SyncRunner.ShellArgs.VATICLE_REPO_PREFIX
+import picocli.CommandLine
+import java.io.File
+import java.nio.file.Path
+import java.util.concurrent.Callable
+import kotlin.io.path.Path
+import kotlin.system.exitProcess
+
+fun main(args: Array<String>): Unit = exitProcess(CommandLine(SyncRunner()).execute(*args))
+
+@CommandLine.Command(name = "sync", mixinStandardHelpOptions = true)
+class SyncRunner : Callable<Unit> {
+
+    @CommandLine.Option(names = ["--verbose", "-v"], required = false)
```

Comment:
can some of these variables be private?

Reply:
Yes, they should all be private.

Getting too comfortable in Rust where things are private by default :(